docs/design/features/covariant-return-methods.md

```diff
@@ -0,0 +1,96 @@
+# Covariant Return Methods
+
+Covariant return methods is a runtime feature designed to support the [covariant return types](https://github.com/dotnet/csharplang/blob/master/proposals/covariant-returns.md) and [records](https://github.com/dotnet/csharplang/blob/master/proposals/records.md) C# language features posed for C# 9.0.
+
+This feature allows an overriding method to have a return type that is different than the one on the method it overrides, but compatible with it. The type compability rules are defined in ECMA I.8.7.1. Example: using a more derived return type. The only exception applicable to this rule for this feature is the compatibility between a value type and an interface that it implements, and this is due to the difference in ABI between reference types and value types (unboxed value types can be returned through a hidden return buffer parameter).
```

Comment:
>exception [](start = 254, length = 9)

This is not an exception to ECMA I.8.7.1. The conversion from a value type to an interface it implements is not included in the *compatible-with* relation defined there.

Reply:
Right. This is more of an implementation detail with the use of `CanCastTo()`. I'll reword it.

Reply:
Honestly, I prefer the reference to the specification. You're welcome to also describe the implementation strategy, but to author metadata/IL we need to know the specification.

---
In reply to: [418240009](https://github.com/dotnet/runtime/pull/35308#discussion_r418240009) [](ancestors = 418240009)